From 578f370c7e7a9f056aefa062b34590b0aa13bce5 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Tue, 15 Aug 2017 17:27:30 +0800
Subject: [PATCH] Fix build on aarch64/musl

Errors

invalid operands to binary & (have 'long double' and 'unsigned int')

Rebase to 0.170
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

---
 backends/aarch64_initreg.c | 1 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/backends/aarch64_initreg.c
+++ b/backends/aarch64_initreg.c
@@ -115,7 +115,7 @@ aarch64_set_initial_registers_tid (pid_t
 
   Dwarf_Word dwarf_fregs[32];
   for (int r = 0; r < 32; r++)
-    dwarf_fregs[r] = fregs.vregs[r] & 0xFFFFFFFF;
+    dwarf_fregs[r] = (unsigned int)fregs.vregs[r] & 0xFFFFFFFF;
 
   if (! setfunc (64, 32, dwarf_fregs, arg))
     return false;
